server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -274,8 +275,7 @@ public void createSegmentTable(final String tableName)
                 + ")",
                 tableName, getPayloadType(), getQuoteString()
             ),
-            StringUtils.format("CREATE INDEX idx_%1$s_datasource ON %1$s(dataSource)", tableName),
```

Comment:
Why these indexes are removed?

Reply:
The index for `datasource` is removed because the new index's prefix covers it already. As for the `used` index, correct me if I am wrong, it does not help any known queries by itself.